### PR TITLE
[REVIEW] Handle `0-dim` inputs while broadcasting to a column

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #2206 

This PR adds support for assigning 0-dim values to a DataFrame, so the fix here is actually needed in detecting a scalar with `is_scalar` method and `to_cudf_compatible_scalar` to get the scalar value from the 0-dim array.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/utils/utils.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def check_equals_int(a, b):
 
 
 def scalar_broadcast_to(scalar, size, dtype=None):
-    from cudf.utils.dtypes import to_cudf_compatible_scalar, is_string_dtype
+    from cudf.utils.dtypes import to_cudf_compatible_scalar
     from cudf.core.column import column_empty
```
Comment (kkraus14) [inline python/cudf/cudf/utils/utils.py:62]: Move these imports out of the function or just use fully qualified name within the function.

File: python/cudf/cudf/utils/utils.py [hunk 2]
```diff
@@ -72,13 +72,10 @@ def scalar_broadcast_to(scalar, size, dtype=None):
...
-        dtype = "object"
-    else:
-        scalar = to_cudf_compatible_scalar(scalar, dtype=dtype)
-        dtype = scalar.dtype
+    scalar = to_cudf_compatible_scalar(scalar, dtype=dtype)
+    dtype = scalar.dtype
 
-    if np.dtype(dtype) == np.dtype("object"):
+    if np.dtype(dtype).kind in ("O", "U"):
         from cudf.core.column import as_column
```
Comment (kkraus14) [inline python/cudf/cudf/utils/utils.py:79]: Same here

File: python/cudf/cudf/utils/dtypes.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import numbers
```
Comment (kkraus14) [inline python/cudf/cudf/utils/dtypes.py:1]: Add copyright to this file

File: python/cudf/cudf/utils/dtypes.py [hunk 4]
```diff
@@ -161,14 +163,18 @@ def to_cudf_compatible_scalar(val, dtype=None):
...
-    dtype = "str" if is_string_dtype(dtype) else dtype
-
     if not is_scalar(val):
         raise ValueError(
             f"Cannot convert value of type {type(val).__name__} "
             " to cudf scalar"
         )
 
+    if isinstance(val, (np.ndarray, cp.ndarray)) and val.ndim == 0:
+        val = val.item()
```
Comment (kkraus14) [inline python/cudf/cudf/utils/dtypes.py:173]: Future optimization: May want to allow 0-dim cupy arrays to pass through here and to use as the underlying memory for a device scalar.

**Comment by galipremsagar** [comment]
> May also want to have our `is_scalar` function return true for these 0-dim arrays.

Had a discussion offline with @kkraus14 and `is_scalar` now returns `true` for 0-dim arrays.

File: python/cudf/cudf/utils/dtypes.py (hunk 4 above)
Comment (galipremsagar) [reply python/cudf/cudf/utils/dtypes.py:173]: Noted, I suppose the only hurdle would be str types which cupy doesn't support. While we would have to specially handle

**Comment by galipremsagar** [comment]
rerun tests

**Comment by galipremsagar** [comment]
rerun tests

**Pull request merged**